src/Agent.Listener/Configuration/IRSAKeyManager.cs

```diff
@@ -81,6 +117,35 @@ private RSAParametersSerializable()
 
         public RSAParametersSerializable(SerializationInfo information, StreamingContext context)
         {
+            bool hasContainerNameMember = false;
+            bool hasUseCngMember = false;
+            var e = information.GetEnumerator();
+            while (e.MoveNext())
+            {
+                if (e.Name == containerNameMemberName)
+                {
+                    hasContainerNameMember = true;
+                }
+
+                if (e.Name == useCngMemberName)
+                {
+                    hasUseCngMember = true;
+                }
+            }
+
+            _containerName = "";
+            _useCng = false;
+
+            if (hasContainerNameMember)
```

Comment:
The fix for the regression from the previous PR is here -- we need to check if the members exist before calling GetValue